### PR TITLE
[CPM-69] Course API

### PR DESCRIPTION
**Pull request by HongShi10**
Contains methods

- GET

- GET ALL

- POST

- PUT

- DELETE

Note: tests for regulation will be done separately

### CODE REVIEW

**Comment by HongShi10** [comment]
Note: Tests will be written in a separate PR

File: api/src/routes/course-route.js [hunk 1]
```diff
@@ -3,9 +3,197 @@ const router = express.Router();
...
+ *            schema:
+ *              $ref: '#/components/schemas/Course'
+ *      responses:
+ *        "200":
+ *          description: A course schema
+ *          content:
+ *            application/json:
+ *              schema:
+ *                $ref: '#/components/schemas/Course'
+ *        "200":
```
Comment (wryao64) [inline api/src/routes/course-route.js:113]: Should this be 400?

File: api/src/routes/course-route.js [hunk 2]
```diff
@@ -3,9 +3,197 @@ const router = express.Router();
...
+            LOGGER.error(err);
+            res.status(400).json({ msg: "Database error" });
+        } else {
+            LOGGER.info("POST Request Suceeded for /Course/");
+            LOGGER.info(product);
+            res.status(201).json(product);
+        }
+    });
+});
+// Creates a new course
```
Comment (wryao64) [inline api/src/routes/course-route.js:161]: This should be delete

**Review by wryao64 — CHANGES_REQUESTED** [review]
Just a few small changes needed

File: api/src/routes/course-route.js (hunk 1 above)
Comment (wryao64) [reply api/src/routes/course-route.js:113]: Just tried running the api - it doesn't run if this is incorrect

File: api/src/models/course.js [hunk 3]
```diff
@@ -70,23 +71,19 @@ const courseSchema = mongoose.Schema({
...
     },
     restrictions: {
-        type: [mongoose.Schema.Types.ObjectId],
-        ref: "Regulation",
+        type: [Regulation],
         required: false,
     },
     informalEquivalents: {
         type: [mongoose.Schema.Types.ObjectId],
-        ref: "Course",
```
Comment (wryao64) [inline api/src/models/course.js:89]: Add this back - probably still need it as a reference to what object it is

**Review by wryao64 — APPROVED** [review]

**Review by anian100 — APPROVED** [review]
LGTM 👍

**Pull request merged**